src/services/rag_service.py

```
@@ -0,0 +1,154 @@
+# src/services/rag_service.py
+import os
+from typing import List, Dict, Any
```

Ruff (0.14.8)

`typing.List` is deprecated, use `list` instead

(UP035)